tests/server.rs

```diff
@@ -2300,43 +2273,6 @@ async fn http2_service_error_sends_reset_reason() {
     assert_eq!(h2_err.reason(), Some(h2::Reason::INADEQUATE_SECURITY));
 }
 
-#[cfg(feature = "stream")]
-#[test]
-fn http2_body_user_error_sends_reset_reason() {
```

Comment:
Ah, yea this test is needed... Not because of the `wrap_stream`, but that if any body returns an error with an HTTP/2 reset code, that it results in an HTTP/2 `RST_STREAM` frame.

Could use a `StreamBody`, or could stay commented out and we file an issue to write a small `struct ImmediatelyError` type just for this test...

Reply:
I wrote a TODO to consider fixing this, as I was unable to fix it.